raft/src/test/java/org/apache/kafka/raft/FileBasedStateStoreTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft;
+
+import org.apache.kafka.test.TestUtils;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.OptionalInt;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+public class FileBasedStateStoreTest {
+
+    @Test
+    public void testReadWrite() throws IOException {
+        final File f = TestUtils.tempFile();
+        final FileBasedStateStore stateStore = new FileBasedStateStore(f);
+        // Invalid or corrupted file should be removed.
+        assertFalse(f.exists());
+        assertEquals(new ElectionState(), stateStore.readElectionState());
+
+        try {
+            final int epoch = 2;
+            final int leaderId = 1;
+            final int votedId = 5;
+
+            stateStore.write(ElectionState.withElectedLeader(epoch, leaderId));
```

Comment:
nit: I would suggest turning these into separate test cases so that we can give them nice names. That makes it easier to see the cases that are covered.